search-service/src/main/kotlin/com/egm/stellio/search/web/TemporalEntityHandler.kt

```diff
@@ -120,9 +120,22 @@ class TemporalEntityHandler(
             jsonLdEntityWithTemporalValues.contexts
         )
 
-        return ResponseEntity.status(HttpStatus.OK).body(serializeObject(filteredJsonLdEntity.compact()))
+        // WORKAROUND:
+        // the json-ld lib shows empty attributes as empty objects but we need empty arrays as stated in 5.7.4.4
+        // replace empty Maps by empty Lists
+        val postProcessedEntity = replaceEmptyMapsByEmptyLists(filteredJsonLdEntity.compact())
```

Comment:
Maybe you can avoid declaring this variable and put the comments above the replaceEmptyMapsByEmptyLists function.